x-pack/plugins/enterprise_search/public/applications/app_search/components/settings/log_retention/log_retention_logic.ts

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { ELogRetentionOptions, ILogRetention, ILogRetentionServer } from './types';
+import { HttpLogic } from '../../../../shared/http';
+import { flashAPIErrors } from '../../../../shared/flash_messages';
+import { convertLogRetentionFromServerToClient } from './utils/convert_log_retention';
+
+interface ILogRetentionActions {
+  clearLogRetentionUpdating(): { value: boolean };
+  closeModals(): { value: boolean };
+  fetchLogRetention(): { value: boolean };
+  saveLogRetention(
+    option: ELogRetentionOptions,
+    enabled: boolean
+  ): { option: ELogRetentionOptions; enabled: boolean };
+  setOpenModal(option: ELogRetentionOptions): { option: ELogRetentionOptions };
+  toggleLogRetention(option: ELogRetentionOptions): { option: ELogRetentionOptions };
+  updateLogRetention(logRetention: ILogRetention): { logRetention: ILogRetention };
+}
+
+interface ILogRetentionValues {
+  logRetention: ILogRetention | null;
+  logsRetentionUpdating: boolean;
+  openModal: ELogRetentionOptions | null;
```

Comment:
[Feel free to disregard this comment totally, just me thinking out loud] This isn't worth the time to change now I think, but if this were new code from scratch I'd probably make the comment the values were a little hard to read/understand the meaning of at a glance.

`logsRetentionUpdating` would be nice to be update to a bool naming convention, e.g. something like `isUpdating` (it's fine to be a little generic IMO because the context comes from where we're pulling the value from).

`openModal` confused me a bit without context because it reads either as an action or possibly a bool. I think it could be clarified to something like `currentModalOption` maybe?

OK sorry rambling is over now haha

Reply:
Stuff like this is easy to change now and virtually risk free. It's not changing the code flow in any significant way, it's just renaming things.

Reply:
https://github.com/elastic/kibana/pull/82364/commits/9a198711b090cda3cab2cc0bafa0172e329029b9
https://github.com/elastic/kibana/pull/82364/commits/267d2333ff8539ccd6e106607a8a12e78a3d92d6

Reply:
Hm, `openedModal` still reads as a potential bool to me (I know it doesn't have an is/has prefix, but we're not yet consistent w/ that naming in our codebase). I really think something like `modalContent` or `modalOption` makes it clearer that the variable being set is a string/piece of copy rather than an open/close flag. But that's definitely not a blocker, so feel free to merge as-is

Reply:
Yeah I see your point. It's interested because it needs to flag that the modal is opened, and which option it is opened for.

Reply:
Maybe I'll try to switch it in my next PR to `modalOption`, so I don't have to wait for CI again.

Reply:
Oh wait. So it **is** a flag for when the modal is opened? Am I just confused? If so, disregard me totally :dead_inside:

Reply:
It's like:

when openedModal
  is null
    modal is open closed
  is 'api'
    modal is opened for editing the 'api' option
  is 'analytics'
    modal is opened for editing the 'analytics' option

Reply:
🙈 Ahh gotcha! I would personally find super helpful for dev readability to pull out a selector called `isModalOpen` that changes to true/false based on `modalOption`, and use that for their UI flags instead of a null/string var. It's really minor but I think it helps, and hey, vars are cheap!